### PR TITLE
[Fusee] Refactor Kernel patching

### PR DESCRIPTION
**Pull request by m4xw**
Add offset based patching
Add svcControlCodeMemory patches for ver. 5, 6, 7

Reviewer:
@SciresM

### CODE REVIEW

File: fusee/fusee-secondary/src/kernel_patches.c [hunk 1]
```diff
@@ -608,36 +631,44 @@ void package2_patch_kernel(void *_kernel, size_t size, bool is_sd_kernel) {
...
-    /* Apply hooks. */
+    /* Apply hooks and patches. */
     uint8_t *kernel = (uint8_t *)_kernel;
     size_t free_space_offset = kernel_info->free_code_space_offset;
     size_t free_space_size = ((free_space_offset + 0xFFFULL) & ~0xFFFULL) - free_space_offset;
-    for (unsigned int i = 0; i < kernel_info->num_hooks; i++) {
-        size_t hook_size = sizeof(instruction_t) * kernel_info->hooks[i].payload_num_instructions;
-        if (kernel_info->hooks[i].branch_back_offset) {
+    for (unsigned int i = 0; i < kernel_info->num_patches; i++) {
+        if(kernel_info->patches[i].patch_offset) {
```
Comment (SciresM) [inline fusee/fusee-secondary/src/kernel_patches.c]: Add space between if and conditional

File: fusee/fusee-secondary/src/kernel_patches.c [hunk 2]
```diff
@@ -1,5 +1,6 @@
 /*
  * Copyright (c) 2018 Atmosphère-NX
+ * Copyright (c) 2019 m4xw <m4x@m4xw.net>
```
Comment (SciresM) [inline fusee/fusee-secondary/src/kernel_patches.c]: I don't object to this necessarily, but we haven't altered the copyright notices on any other files following pull requests to the project. Do you consider this important?

Comment (m4xw) [reply fusee/fusee-secondary/src/kernel_patches.c]: That's usually what projects do to track contributions (history could be squashed at some point etc.)
Also I am not part of the Atmosphere team, if I need to be contacted for lets say a theoretical license change, my contact information will be present

File: fusee/fusee-secondary/src/kernel_patches.c (hunk 1 above)
Comment (m4xw) [reply fusee/fusee-secondary/src/kernel_patches.c]: Assume you do squash?

**Pull request merged**